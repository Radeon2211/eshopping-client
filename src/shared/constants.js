import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
