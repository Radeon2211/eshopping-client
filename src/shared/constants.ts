import * as Yup from 'yup';
import countryCodes from 'country-codes-list';
import { ProductCondition, ProductPhotoFieldValue } from './types/enums';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([ProductCondition.NEW, ProductCondition.USED, ProductCondition.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
  photo: Yup.mixed()
    .test('is-error', (value) => value !== ProductPhotoFieldValue.ERROR)
    .nullable(),
};

export const listOfAreaCodes = countryCodes.customArray({
  value: '{countryCallingCode}',
  label: '+{countryCallingCode} {countryNameEn}',
});

export const listOfCountries = countryCodes.customArray({
  value: '{countryNameEn}',
  label: '{countryNameEn}',
});

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: 'Price - ascending' },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: 'Price - ascending' },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig: ScrollToOptions = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};
