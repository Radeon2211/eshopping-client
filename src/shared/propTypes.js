import PropTypes from 'prop-types';

export const deliveryAddressStrings = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  phonePrefix: PropTypes.string,
  phoneNumber: PropTypes.string,
  street: PropTypes.string,
  zipCode: PropTypes.string,
  city: PropTypes.string,
  country: PropTypes.string,
};

export const userProfile = {
  _id: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  contacts: PropTypes.shape({
    email: PropTypes.bool.isRequired,
    phone: PropTypes.bool.isRequired,
  }).isRequired,
  ...deliveryAddressStrings,
};

export const fullProductItem = {
  _id: PropTypes.string.isRequired,
  quantitySold: PropTypes.number.isRequired,
  buyerQuantity: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  condition: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  updatedAt: PropTypes.string.isRequired,
  __v: PropTypes.number.isRequired,
  photo: PropTypes.bool.isRequired,
  seller: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }).isRequired,
};

export const cartItem = {
  _id: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
  product: PropTypes.shape(fullProductItem).isRequired,
};

export const orderItem = {
  _id: PropTypes.string.isRequired,
  products: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      quantity: PropTypes.number.isRequired,
      photo: PropTypes.bool.isRequired,
    }),
  ).isRequired,
  seller: PropTypes.shape({
    username: PropTypes.string.isRequired,
    phone: PropTypes.string,
    email: PropTypes.string,
  }),
  buyer: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }),
  deliveryAddress: PropTypes.shape(deliveryAddressStrings).isRequired,
  overallPrice: PropTypes.number.isRequired,
  createdAt: PropTypes.string.isRequired,
  updatedAt: PropTypes.string.isRequired,
  __v: PropTypes.number.isRequired,
};

export const signupErrors = {
  email: PropTypes.string,
  username: PropTypes.string,
  password: PropTypes.string,
  ...deliveryAddressStrings,
};

export const signupTouched = {
  email: PropTypes.bool,
  username: PropTypes.bool,
  password: PropTypes.bool,
  firstName: PropTypes.bool,
  lastName: PropTypes.bool,
  phonePrefix: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
  phoneNumber: PropTypes.bool,
  street: PropTypes.bool,
  zipCode: PropTypes.bool,
  city: PropTypes.bool,
  country: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
  'react-select-3-input': PropTypes.bool,
  'react-select-4-input': PropTypes.bool,
};

export const signupValues = {
  email: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  phonePrefix: PropTypes.shape({
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
  }),
  phoneNumber: PropTypes.string.isRequired,
  street: PropTypes.string.isRequired,
  zipCode: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  country: PropTypes.shape({
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
  }),
};
