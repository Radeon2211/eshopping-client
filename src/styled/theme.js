const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
