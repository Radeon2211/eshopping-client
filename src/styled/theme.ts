const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#777',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#06AD96',
    greenDark: '#009680',
    red: '#F22613',
    redLight: '#F94D2A',
    redDark: '#985A5A',
  },
  shadows: {
    level1: '0 0 3px rgba(0, 0, 0, 0.03)',
    level2: '0 0 3px rgba(0, 0, 0, 0.3)',
    level3: '0 0 5px rgba(0, 0, 0, 0.3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Noto Sans, sans-serif',
    text: 'Nunito, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
    level5: 100000,
  },
  fontSizes: {
    level1: '1.3rem',
    level2: '1.4rem',
    level3: '1.5rem',
    level4: '1.7rem',
    level5: '2.2rem',
    level6: '2.6rem',
  },
  durations: {
    level1: 0.1,
    level2: 0.2,
    level3: 0.3,
  },
  lineHeights: {
    level1: '1',
    level2: '1.1',
    level3: '1.2',
    level4: '1.3',
    level5: '1.4',
    level6: '1.5',
  },
};

export type ThemeSpacing = keyof typeof theme.spacings;
export type ThemeZIndex = keyof typeof theme.zIndexes;
export type ThemeFontSize = keyof typeof theme.fontSizes;
export type ThemeDuration = keyof typeof theme.durations;
export type ThemeLineHeight = keyof typeof theme.lineHeights;

export default theme;
