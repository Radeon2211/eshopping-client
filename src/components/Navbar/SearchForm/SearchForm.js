import React, { useState, useEffect, useCallback } from 'react';
import { useHistory } from 'react-router-dom';
import * as SC from './SearchForm.sc';
import Button from '../../UI/Button/Button';
import MyIcon from '../../UI/MyIcon';
import { ReactComponent as SearchIcon } from '../../../images/icons/search.svg';
import { defaultAppPath } from '../../../shared/constants';
import { getParamsWithoutPollution } from '../../../shared/utility/utility';
import useWindowSize from '../../../shared/useWindowSize';

export default function SearchForm() {
  const [productName, setProductName] = useState('');
  const history = useHistory();

  const windowSize = useWindowSize();

  const checkQueryParams = useCallback(
    (queryParams) => {
      const correctQueryParams = getParamsWithoutPollution(queryParams);
      const { name = '' } = correctQueryParams;
      setProductName(name);
    },
    [setProductName],
  );

  useEffect(() => {
    checkQueryParams(history.location.search);
    history.listen(({ search }) => {
      checkQueryParams(search);
    });
  }, [history, setProductName, checkQueryParams]);

  const formSubmitHandle = (e) => {
    e.preventDefault();
    const correctQueryParams = getParamsWithoutPollution(history.location.search);
    const { name = '' } = correctQueryParams;
    if (name === productName) return;
    let nameParam = '';
    if (productName.length > 0) {
      nameParam = `&name=${productName}`;
    }
    history.push(`${defaultAppPath}${nameParam}`);
  };

  let buttonContent = 'search';
  if (windowSize.width < 900) {
    buttonContent = (
      <MyIcon size="small" color="#fff">
        <SearchIcon />
      </MyIcon>
    );
  }

  return (
    <SC.SearchForm onSubmit={formSubmitHandle}>
      <input
        type="text"
        className="name-input"
        name="product"
        value={productName}
        maxLength="150"
        autoComplete="off"
        spellCheck="false"
        onChange={(e) => setProductName(e.target.value)}
        data-testid="SearchForm-input"
      />
      <Button type="submit" filled aria-label="Search">
        {buttonContent}
      </Button>
    </SC.SearchForm>
  );
}
