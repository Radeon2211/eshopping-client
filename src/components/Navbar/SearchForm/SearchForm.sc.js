import styled from 'styled-components';

export const SearchForm = styled.form`
  display: flex;
  max-width: 100%;
  width: 45rem;

  & .input-text {
    border: 1px solid ${({ theme }) => theme.colors.light3};
    font-size: 1.5rem;
    outline: none;
    padding: ${({ theme }) => theme.spacings.level1};
    width: 100%;
  }

  @media only screen and (max-width: 37.5em) {
    order: 2;
  }
`;