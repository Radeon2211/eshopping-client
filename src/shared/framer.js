import theme from '../styled/theme';

export const stepFormVariants = {
  hidden: {
    opacity: 0,
  },
  visible: {
    opacity: 1,
    transition: { duration: theme.durations.level1, delay: theme.durations.level1 },
  },
};

export const backdropVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    transition: { duration: theme.durations.level3 },
  },
};

export const modalVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    transition: { delay: theme.durations.level2 },
    pointerEvents: 'initial',
  },
};

export const messageBoxVariants = {
  hidden: {
    pointerEvents: 'none',
    y: '-150%',
  },
  visible: {
    transition: { type: 'spring', duration: theme.durations.level2, stiffness: 100 },
    pointerEvents: 'initial',
    y: 0,
  },
};

export const dropdownVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
    y: '-10%',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    y: '0',
  },
};
