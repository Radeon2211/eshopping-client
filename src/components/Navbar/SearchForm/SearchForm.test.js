import React from 'react';
import { mount } from 'enzyme';
import { Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import SearchForm from './SearchForm';
import * as SC from './SearchForm.sc';
import theme from '../../../styled/theme';
import { defaultAppPath } from '../../../shared/constants';

const setUp = (push = jest.fn(), search = '?p=1') => {
  const history = {
    listen: jest.fn(),
    createHref: jest.fn(),
    location: { pathname: '/products', search },
    push,
  };
  return mount(
    <Router history={history}>
      <ThemeProvider theme={theme}>
        <SearchForm />
      </ThemeProvider>
    </Router>,
  );
};

describe('<SearchForm />', () => {
  describe(`Check if correct url is pushing to history`, () => {
    it('should push defaultAppPath&name=test-name if there is no name in url', () => {
      const pushFn = jest.fn();
      const wrapper = setUp(pushFn);
      const form = wrapper.find(SC.SearchForm);
      const nameInput = wrapper.find('.name-input');
      nameInput.simulate('change', { target: { value: 'test-name' } });
      form.simulate('submit');
      expect(pushFn).toBeCalledWith(`${defaultAppPath}&name=test-name`);
    });

    it('should push defaultAppPath if in url is name', () => {
      const pushFn = jest.fn();
      const wrapper = setUp(pushFn, '?p=1&name=test-name');
      const form = wrapper.find(SC.SearchForm);
      const nameInput = wrapper.find('.name-input');
      nameInput.simulate('change', { target: { value: '' } });
      form.simulate('submit');
      expect(pushFn).toBeCalledWith(defaultAppPath);
    });

    it('should NOT call push if in url is no name and input is empty', () => {
      const pushFn = jest.fn();
      const wrapper = setUp(pushFn);
      const form = wrapper.find(SC.SearchForm);
      form.simulate('submit');
      expect(pushFn).not.toBeCalled();
    });

    it('should NOT call push if in url is name and input is not edited', () => {
      const pushFn = jest.fn();
      const wrapper = setUp(pushFn, '?p=1&name=test-name');
      const form = wrapper.find(SC.SearchForm);
      form.simulate('submit');
      expect(pushFn).not.toBeCalled();
    });
  });
});
