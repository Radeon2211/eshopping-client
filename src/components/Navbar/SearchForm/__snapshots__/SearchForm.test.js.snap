// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> check how renders should render with icon in button if width is lower than 900px 1`] = `
<DocumentFragment>
  <form
    class="sc-bdfBwQ eVTBHG"
  >
    <input
      autocomplete="off"
      class="name-input"
      data-testid="SearchForm-input"
      maxlength="150"
      name="product"
      spellcheck="false"
      type="text"
      value=""
    />
    <button
      aria-label="Search"
      class="sc-gsTCUz fUKWKF"
      color="blue"
      type="submit"
    >
      <div
        class="sc-gKsewC zQbGG"
        color="#fff"
      >
        <svg>
          search.svg
        </svg>
      </div>
    </button>
  </form>
</DocumentFragment>
`;

exports[`<SearchForm /> check how renders should render with text in button if width is equal or greater than 900px 1`] = `
<DocumentFragment>
  <form
    class="sc-bdfBwQ eVTBHG"
  >
    <input
      autocomplete="off"
      class="name-input"
      data-testid="SearchForm-input"
      maxlength="150"
      name="product"
      spellcheck="false"
      type="text"
      value=""
    />
    <button
      aria-label="Search"
      class="sc-gsTCUz fUKWKF"
      color="blue"
      type="submit"
    >
      search
    </button>
  </form>
</DocumentFragment>
`;
